src/Host/Client/Impl/Host/BrokerClientProxy.cs

```diff
@@ -34,11 +34,10 @@ internal sealed class BrokerClientProxy : IBrokerClient {
 
         public Task PingAsync() => _broker.PingAsync();
 
-        public async Task<RHost> ConnectAsync(string name, IRCallbacks callbacks, string rCommandLineArguments = null, int timeout = 3000,
-            CancellationToken cancellationToken = default(CancellationToken), ReentrancyToken reentrancyToken = default(ReentrancyToken)) {
+        public async Task<RHost> ConnectAsync(string name, IRCallbacks callbacks, string rCommandLineArguments = null, int timeout = 3000, bool preserveSessionData = false, CancellationToken cancellationToken = default(CancellationToken), ReentrancyToken reentrancyToken = default(ReentrancyToken)) {
```

Comment:
Sure. Another thing that I'm considering is to use `AsyncLocal` instead of `ReentrancyToken`. That will simplify signature, the only problem is that `AsyncLocal` pollutes `ExecutionContext` which is kind of expensive operation (any write to the `AsyncLocal` creates a new copy of the dictionary that contains all values).